Update docs with an image and some small corrections

paytonrules opened the pull request.

paytonrules committed:
Update docs with an image and some small corrections

rmulhol commented on transformers/events/DOCUMENTATION.md:
```diff
@@ -2,11 +2,13 @@
 
 ## Architecture
 
-Transformers fetch logs from Ethereum, convert/decode them into usable data, and then persist them in postgres. A transformer converts the raw chain data into a human friendly representation suitable for consumption in the API.
```
Wondering if we might want to keep something stating that configuring event transformers in execute causes us to query the ethereum node for those logs and write them to the `public.event_logs` table (in addition to scanning that table and transforming them)

paytonrules replied:
I did not know it did that - yes it deserves a mention.

paytonrules replied:
I'm not sure what happened with the top here, I basically restored the top paragraph as I like it better. Blame turkey.

rmulhol commented on transformers/events/DOCUMENTATION.md:
```diff
@@ -29,28 +31,27 @@ If the contract isn't already present in the environment you'll need to add it b
+Much of the time it is possible to complete this step without creating the database
+migration, because to write the transformer you don't actually save the transformed
 object to the database. You may be saving addresses, which
 go into the already existing address table and are referenced by a foreign key. 
```
Hm, I'm confused by this line. I think every transformer writes to a new table? Even if the table is functionally just a join table of `log_id` + `address_id`, I think it's normally declared on a table matching the event name

paytonrules replied:
You don't have to do it _yet_ to make the test pass. You'll need it for the integration test.

rmulhol replied:
I think maybe I'm thrown off by the line about "You may be saving an address" - the reason you don't need a migration isn't because of the content of the table, it's because the unit tests for individual transformers don't exercise writing the new record to the DB (that only happens in integration tests), right?

paytonrules replied:
That's correct - also that's not new. It's how we've phrased it for a while.

rmulhol commented on transformers/events/DOCUMENTATION.md:
```diff
@@ -168,3 +169,9 @@ In the event there are not logs for an event you're looking to transform in ethe
+
+This shows the entire process, from the perspective of the `execute` process. Writing transformers means writing code for the little loop near the end.
+
+![Transformer Sequence Diagram](./Transforming Events.svg)
```
do we need to include this file in this PR?

paytonrules replied:
Only if we want it to show up.

paytonrules committed:
Add the image of the transformer sequence diagram

paytonrules committed:
Fix up some comments in the documentation

Clarify that you need the database table eventually, and that
public.event_logs are updated.

paytonrules committed:
Rename Transforming Events so there is no space

This fixes rendering the image, hopefully.

rmulhol reviewed (approved):
Good with this as is, definite improvement over what we had before 👍 

Still think it's a little confusing to imply folks don't need a migration to add a new table for the new transformed event, but we can always shore that up later
